apps/src/code-studio/pd/workshop_dashboard/components/edit_enrollment_name_dialog.jsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Modal, Button, Row, Col, FormGroup} from 'react-bootstrap';
+import _ from 'lodash';
+
+export default class EditEnrollmentNameDialog extends React.Component {
+  static propTypes = {
+    selectedEnrollment: PropTypes.object,
+    show: PropTypes.bool,
+    onShow: PropTypes.func,
+    onCancel: PropTypes.func,
+    onEdit: PropTypes.func
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      firstName: '',
+      lastName: ''
+    };
+
+    this.handleInputChange = this.handleInputChange.bind(this);
```

Comment:
I don't understand what this does (modeled off "Move enrollment" dialog, if anyone can enlighten me.

Reply:
`bind` actually creates a new function which has its `this` keyword set to the provided value.  That way it knows which `this` it's working with.  

You'll see in ES6 code that it's becoming more common to just call the function via `() =>` in the event handler, and it will take care of calling it with the right `this` context.

Reply:
That makes sense I think -- I would have thought that the `this` for `handleInputChange` would already be the component's `this`, which is what it seems like it's doing? What am I missing?

Reply:
ooh maybe I understand...`this` by default in this case would be set to the html element from which these methods get called (the `input` elements). Instead, we're setting `this` to be the `this` of the component?

Reply:
According to this page, `this` will be `undefined`: https://reactjs.org/docs/handling-events.html.

Reply:
Perfect, just what I was looking for thanks!